todo app/src/store/todo.js

```diff
@@ -0,0 +1,34 @@
+import { createSlice } from "@reduxjs/toolkit";
+
+const todoSlice = createSlice({
+  name: "Todo",
+  initialState: [
+    {
+      id: "0",
+      task: "IAmHere",
+      done: false,
+    },
+  ],
+  reducers: {
+    add(state, action) {
+      const text = action.payload.trim();
+      if (text.length === 0) return;
```

Comment:
This is small but I would recommend putting validations out of the reducer. Usually, it is recommended to put as much logic as possible in the reducers but logic related to new state calculation. Maybe put the validation in the component itself so that we don't even have to call the reducer when the data is not valid

Reply:
Another thing that I thinking about (but not sure if that's the case) is that whatever is returned by the reducer will be treated as the new state, therefore by returning `undefined` you would be overwriting the previous state to be undefined which is not the desired behavior I guess.

Reply:
Yes, for redux whenever you return the returned data overwrites the current state. But for redux toolkit it whenever you mutate the current state in the background an updated state is created that's why we no longer needs to return.
In my case since I returned without mutating the state the state stays the same (not undefined)